File: backend/controllers/audit_controller.go
package controllers

import (
	"net/http"
	"time"

	"governanca-ti/config"
	"governanca-ti/models"

	"github.com/gin-gonic/gin"
)

// GetAuditLogs busca o histórico de ações no sistema
func GetAuditLogs(c *gin.Context) {
	var logs []models.AuditLog
	
	// Busca os logs ordenados do mais recente para o mais antigo (limitado aos últimos 200 para não pesar o painel)
	if err := config.DB.Order("timestamp desc").Limit(200).Find(&logs).Error; err != nil {
		c.JSON(http.StatusInternalServerError, gin.H{"error": "Erro ao buscar logs"})
		return
	}

	c.JSON(http.StatusOK, gin.H{"data": logs})
}

// CreateAuditLog registra uma nova ação feita por um usuário
func CreateAuditLog(c *gin.Context) {
	var input models.AuditLog
	
	if err := c.ShouldBindJSON(&input); err != nil {
		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
		return
	}

	// Garante que a data/hora do log seja exata no momento do registro
	input.Timestamp = time.Now()

	if err := config.DB.Create(&input).Error; err != nil {
		c.JSON(http.StatusInternalServerError, gin.H{"error": "Erro ao salvar o log no banco"})
		return
	}

	c.JSON(http.StatusCreated, gin.H{"message": "Log registrado com sucesso"})
}

File: backend-node/src/controllers/auditController.js
// ✅ Importação centralizada no db.js
const { AuditLog } = require('../../config/db');

exports.getAll = async (req, res) => {
  try {
    // Busca os logs ordenados do mais recente para o mais antigo 
    const logs = await AuditLog.findAll({
      order: [['createdAt', 'DESC']],
      limit: 200 // Proteção de performance para a Hostinger
    });

    // Retorna no formato data: [] para o seu Axios no React
    return res.status(200).json({ data: logs });
  } catch (error) {
    console.error("❌ Erro ao buscar logs de auditoria:", error.message);
    return res.status(500).json({ error: 'Erro ao buscar logs no banco de dados' });
  }
};

exports.create = async (req, res) => {
  try {
    const input = req.body;

    // O Sequelize cria o registro e preenche o 'createdAt' automaticamente
    const newLog = await AuditLog.create(input);

    return res.status(201).json({ 
      message: 'Log registrado com sucesso',
      data: newLog 
    });
  } catch (error) {
    console.error("❌ Erro ao registrar log:", error.message);
    return res.status(400).json({ error: 'Falha ao salvar o log de auditoria' });
  }
};